### PR TITLE
BUG: tools/cgxset: Fix potential memory leak in converted_src_cgroup

### PR DESCRIPTION
**Pull request by drakenclimber**
There's a brief window where converted_src_cgroup is allocated but hasn't been assigned to the *cgroup pointer.  If a failure occurs in this window, then converted_src_cgroup must be freed before goto'ing to the error path.

### CODE REVIEW

File: src/tools/cgxset.c [hunk 1]
```diff
@@ -289,6 +289,11 @@ int main(int argc, char *argv[])
 			goto err;
```
Comment (kamalesh-babulal) [inline src/tools/cgxset.c:289]: @drakenclimber good catch.  One improvement would be to replace the goto `err`  with goto `cgroup_free_err`.   The `cgroup` is allocated and needs to free'd too.

File: src/tools/cgxset.c [hunk 2]
```diff
@@ -298,8 +303,14 @@ int main(int argc, char *argv[])
 			 * v1 to v2 or vice versa
 			 */
 			ret = 0;
-		else if (ret)
+		else if (ret) {
+			free(converted_src_cgroup);
 			goto err;
```
Comment (kamalesh-babulal) [inline src/tools/cgxset.c:308]: This goto `err` too needs to be replaced with `cgroup_free_err`

Comment (drakenclimber) [reply src/tools/cgxset.c:308]: Good catch.  Looking through the function, I don't see the point of the `err` tag.  I propose we delete the `err` tag and have all the goto's go to `cgroup_free_err`.  Thoughts?

Comment (drakenclimber) [reply src/tools/cgxset.c:308]: It was lazier (easier) to delete the `cgroup_free_tag` and have everything goto `err`.  Let me know what you think.

**Comment by drakenclimber** [comment]
@kamalesh-babulal - Would you mind taking a quick look at my latest changes and make sure I didn't screw anything up?  ;)  Thanks!

File: src/tools/cgxset.c (hunk 2 above)
Comment (kamalesh-babulal) [reply src/tools/cgxset.c:308]: > Looking through the function, I don't see the point of the err tag. I propose we delete the err tag and have all the goto's go to cgroup_free_err. Thoughts?

Agreed, there is no point in having an extra label for the error path, given that there are checks for their validity before calling `cgroup_free()` and at the maximum, extra if checks will be performed in the error path.

**Comment by kamalesh-babulal** [comment]
> Would you mind taking a quick look at my latest changes and make sure I didn't screw anything up? ;) Thanks!

@drakenclimber the changes look good to me and apologies for missing the update.

**Review by kamalesh-babulal — APPROVED** [review]

**Comment by drakenclimber** [comment]
Pushed to main in commit f1579cec2734.  Pushed to release-3.0 in commit 8da741e19bb5.  Thanks, @kamalesh-babulal

**Pull request closed without merging**